### PR TITLE
development of unit test case

### PR DESCRIPTION
**Pull request by RahulPIO**

### CODE REVIEW

File: src/test/java/com/pio/security/jwt/repository/UserRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@SpringBootTest
+class UserRepositoryTest {
+    @Autowired
+    private UserRepository userRepository;
+
+    @Test
+    void testSave() {
```
Comment (vaibhavPio) [inline src/test/java/com/pio/security/jwt/repository/UserRepositoryTest.java]: use more descriptive method names like
shouldSaveUserSuccessfully() and shouldFindUserByUsername() 
to make test purposes clearer.

File: src/test/java/com/pio/security/jwt/repository/UserRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        user.setPassword("J1234");
+        user.setRole(UserRole.USER);
+        UserEntity resultUser = userRepository.save(user);
+        assertEquals(user.getUsername(),resultUser.getUsername());
+        assertEquals(user.getPassword(),resultUser.getPassword());
+    }
+
+    @Test
+    void testFindByUsername() {
+        UserEntity user = new UserEntity();
```
Comment (vaibhavPio) [inline src/test/java/com/pio/security/jwt/repository/UserRepositoryTest.java]: save and findById both methods create a UserEntity with similar setup,
you can extract this user creation logic into helper method to improve readability and 
reduce duplication.

File: src/test/java/com/pio/security/jwt/security/JwtHelperTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Mock
+    private JwtHelper jwtHelper;
+
+    @Mock
+    private UserDetails userDetails;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.initMocks(this);
```
Comment (vaibhavPio) [inline src/test/java/com/pio/security/jwt/security/JwtHelperTest.java]: you already have @ExtendWith(MockitoExtension.class) in newer version of Mockito initMocks() is deprecated and unnecessary it can be removed.

File: src/test/java/com/pio/security/jwt/service/UserServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    @BeforeEach
+    void setUp() {
+        userDTO = new UserDTO();
+        userDTO.setUsername("abc");
+        userDTO.setPassword("1234");
+        userDTO.setRole(UserRole.ADMIN);
+    }
+
+    @Test
+    void testAddUserSuccess() {
```
Comment (vaibhavPio) [inline src/test/java/com/pio/security/jwt/service/UserServiceTest.java]: you could add more edge case, such as null or empty username/password, to make test more robust.

File: src/test/java/com/pio/security/jwt/service/UserServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        UserNotFoundException exceptionThrown = assertThrows(UserNotFoundException.class, () ->{
+                    userService.getUserById(1);
+                });
+        assertEquals(404,exceptionThrown.getHttpStatus());
+        assertEquals("User not found with id " +1,exceptionThrown.getMessage());
+    }
+
+    @Test
+    void testGetAllUserSuccess() {
+        List<UserEntity> userEntityList = Arrays.asList(new UserEntity(1,"abc","123",UserRole.USER),new UserEntity(2,"xyz","789",UserRole.USER));
```
Comment (vaibhavPio) [inline src/test/java/com/pio/security/jwt/service/UserServiceTest.java]: Avoid hardcoding string like "abc", "1234", "Username already taken". 
Use constants or builder classes to improve clarity and maintainability.

File: src/test/java/com/pio/security/jwt/repository/UserRepositoryTest.java (hunk 2 above)
Comment (RahulPIO) [reply src/test/java/com/pio/security/jwt/repository/UserRepositoryTest.java]: Resolved

File: src/test/java/com/pio/security/jwt/security/JwtHelperTest.java (hunk 3 above)
Comment (RahulPIO) [reply src/test/java/com/pio/security/jwt/security/JwtHelperTest.java]: resolved

File: src/main/java/com/pio/security/jwt/service/UserService.java [hunk 6]
```diff
@@ -59,10 +59,15 @@ public void addUser(UserDTO userDTO) {
         if (optionalUser.isPresent()) {
             throw new UserAlreadyExistsException(400, "Username already taken");
```
Comment (harshitmishra-pio) [inline src/main/java/com/pio/security/jwt/service/UserService.java]: Error code and messages must come from constant files respectively

File: src/main/java/com/pio/security/jwt/service/UserService.java [hunk 7]
```diff
@@ -59,10 +59,15 @@ public void addUser(UserDTO userDTO) {
...
+        saveUser(userDTO);
+    }
+
+    private String saveUser(UserDTO userDTO) {
         UserEntity userEntity = new UserEntity();
         userEntity.setPassword(passwordEncoder.encode(userDTO.getPassword()));
         userEntity.setUsername(userDTO.getUsername());
         userEntity.setRole(userDTO.getRole());
         userRepository.save(userEntity);
+        return "User Added Successfully";
```
Comment (harshitmishra-pio) [inline src/main/java/com/pio/security/jwt/service/UserService.java]: Please send proper response.

**Review by harshitmishra-pio — CHANGES_REQUESTED** [review]
